From: Dmitry Shachnev <mitya57@debian.org>
Date: Wed, 25 Dec 2019 23:18:56 +0300
Subject: Remove references to sphinxcontrib modules

They are not packaged yet, and when they are, we would want to avoid
dependency loops anyway.
---
 doc/usage/builders/index.rst     | 24 ------------------------
 sphinx/application.py            |  6 ------
 sphinx/builders/html/__init__.py |  2 --
 3 files changed, 32 deletions(-)

diff --git a/doc/usage/builders/index.rst b/doc/usage/builders/index.rst
index 4d53152..4f7bdb6 100644
--- a/doc/usage/builders/index.rst
+++ b/doc/usage/builders/index.rst
@@ -68,12 +68,6 @@ The builder's "name" must be given to the **-b** command-line option of
    also generates HTML Help support files that allow the Microsoft HTML Help
    Workshop to compile them into a CHM file.
 
-   .. autoattribute:: name
-
-   .. autoattribute:: format
-
-   .. autoattribute:: supported_image_types
-
 .. module:: sphinxcontrib.qthelp
 .. class:: QtHelpBuilder
 
@@ -85,12 +79,6 @@ The builder's "name" must be given to the **-b** command-line option of
 
       Moved to sphinxcontrib.qthelp from sphinx.builders package.
 
-   .. autoattribute:: name
-
-   .. autoattribute:: format
-
-   .. autoattribute:: supported_image_types
-
    .. _Qt help: https://doc.qt.io/qt-4.8/qthelp-framework.html
 
 .. module:: sphinxcontrib.applehelp
@@ -111,12 +99,6 @@ The builder's "name" must be given to the **-b** command-line option of
    output will not be valid until :program:`hiutil` has been run on all of the
    ``.lproj`` folders within the bundle.
 
-   .. autoattribute:: name
-
-   .. autoattribute:: format
-
-   .. autoattribute:: supported_image_types
-
    .. versionadded:: 1.3
 
    .. versionchanged:: 2.0
@@ -130,12 +112,6 @@ The builder's "name" must be given to the **-b** command-line option of
    also generates `GNOME Devhelp <https://wiki.gnome.org/Apps/Devhelp>`__
    support file that allows the GNOME Devhelp reader to view them.
 
-   .. autoattribute:: name
-
-   .. autoattribute:: format
-
-   .. autoattribute:: supported_image_types
-
    .. versionchanged:: 2.0
 
       Moved to sphinxcontrib.devhelp from sphinx.builders package.
diff --git a/sphinx/application.py b/sphinx/application.py
index b55eb76..399ce14 100644
--- a/sphinx/application.py
+++ b/sphinx/application.py
@@ -107,12 +107,6 @@ builtin_extensions = (
     'sphinx.environment.collectors.metadata',
     'sphinx.environment.collectors.title',
     'sphinx.environment.collectors.toctree',
-    # 1st party extensions
-    'sphinxcontrib.applehelp',
-    'sphinxcontrib.devhelp',
-    'sphinxcontrib.htmlhelp',
-    'sphinxcontrib.serializinghtml',
-    'sphinxcontrib.qthelp',
     # Strictly, alabaster theme is not a builtin extension,
     # but it is loaded automatically to use it as default theme.
     'alabaster',
diff --git a/sphinx/builders/html/__init__.py b/sphinx/builders/html/__init__.py
index 6e73253..b461a25 100644
--- a/sphinx/builders/html/__init__.py
+++ b/sphinx/builders/html/__init__.py
@@ -1299,8 +1299,6 @@ def migrate_html_add_permalinks(app: Sphinx, config: Config) -> None:
     )
 
 # for compatibility
-import sphinxcontrib.serializinghtml  # NOQA
-
 import sphinx.builders.dirhtml  # NOQA
 import sphinx.builders.singlehtml  # NOQA
 
